nemo/collections/asr/data/audio_to_text_lhotse_speaker.py

```diff
@@ -0,0 +1,163 @@
+        return {
+            'audio_signal': NeuralType(('B', 'T'), AudioSignal()),
+            'a_sig_length': NeuralType(tuple('B'), LengthsType()),
+            'transcripts': NeuralType(('B', 'T'), LabelsType()),
+            'transcript_length': NeuralType(tuple('B'), LengthsType()),
+            'spk_tar_id': NeuralType(('B','T'), LabelsType()),
+            'sample_id': NeuralType(tuple('B'), LengthsType(), optional=True),
+        }
+
+    def __init__(self, cfg, tokenizer):
+        super().__init__()
+        self.tokenizer = TokenizerWrapper(tokenizer)
+        self.load_audio = AudioSamples(fault_tolerant=True)
+        self.cfg = cfg
+        self.rttms = SupervisionSet.from_rttm(Path(self.cfg.rttm_filepath).rglob('*.rttm'))
+        self.spk_tar_all_zero = self.cfg.get('spk_tar_all_zero',False)
+        self.num_speakers = self.cfg.get('num_speakers', 4)
+        self.num_sample_per_mel_frame = self.cfg.get('num_sample_per_mel_frame', 160)
+        self.num_mel_frame_per_asr_frame = self.cfg.get('num_mel_frame_per_asr_frame', 8)
+        
+
+    def __getitem__(self, cuts) -> Tuple[torch.Tensor, ...]:
+        spk_targets = [torch.transpose(torch.as_tensor(self.speaker_to_target(c, self.num_speakers, self.num_sample_per_mel_frame, self.num_mel_frame_per_asr_frame, self.spk_tar_all_zero), dtype=torch.float32), 0, 1) for c in cuts]
+        audio, audio_lens, cuts = self.load_audio(cuts)
+        tokens = [torch.as_tensor(self.tokenizer(c.supervisions[0].text, c.supervisions[0].language)) for c in cuts]
+        token_lens = torch.tensor([t.size(0) for t in tokens], dtype=torch.long)
+        tokens = collate_vectors(tokens, padding_value=0)
+        spk_targets = collate_matrices(spk_targets)
+
+
+        return audio, audio_lens, tokens, token_lens, spk_targets
+    
+
+    def speaker_to_target(self, cut, num_speakers: int = 4, num_sample_per_mel_frame: int = 160, num_mel_frame_per_asr_frame: int = 8, spk_tar_all_zero: bool = False):
+        '''
+        get rttm samples corresponding to one cut, generate speaker_audio_mask numpy.ndarray
+        '''
+
+        #get cut-related segments from rttms
+        segments_iterator = self.rttms.find(recording_id=cut.recording_id, start_after=cut.start, end_before=cut.end, adjust_offset=True)
+
+        segments = [s for s in segments_iterator]
+
+        #generate speaker id according to arrival time
+        segments.sort(key = lambda rttm_sup: rttm_sup.start)
+
+        seen = set()
+        seen_add = seen.add
+        speaker_ats = [s.speaker for s in segments if not (s.speaker in seen or seen_add(s.speaker))]
+        
+        speaker_to_idx_map = {
+                spk: idx
+                for idx, spk in enumerate(speaker_ats)
+        }
+
+        
+        #initialize mask matrices (num_speaker, encoder_hidden)
+        mask = np.zeros((num_speakers, int(np.ceil(np.ceil((cut.num_samples + 1) / num_sample_per_mel_frame) / num_mel_frame_per_asr_frame))))
```

Comment:
This is better to be a standalone static function: 

@staticmethod
def get_frame_length_from_sample_length(num_samples):
 ...